node/src/accountant/mod.rs

```diff
@@ -787,34 +817,59 @@ impl Accountant {
         let msg_id = self.msg_id();
         trace!(
             self.logger,
-            "MsgId {}: Accruing debt to {} for consuming {} exited bytes",
+            "Msg {}: Accruing debt to {} for consuming {} exited bytes",
             msg_id,
             msg.exit.earning_wallet,
             msg.exit.payload_size
         );
-        self.record_service_consumed(
+        let exit_sum = self.record_service_consumed(
             msg.exit.service_rate,
             msg.exit.byte_rate,
             msg.timestamp,
             msg.exit.payload_size,
             &msg.exit.earning_wallet,
         );
-        msg.routing.iter().for_each(|routing_service| {
-            trace!(
-                self.logger,
-                "MsgId {}: Accruing debt to {} for consuming {} routed bytes",
-                msg_id,
-                routing_service.earning_wallet,
-                msg.routing_payload_size
-            );
-            self.record_service_consumed(
-                routing_service.service_rate,
-                routing_service.byte_rate,
-                msg.timestamp,
-                msg.routing_payload_size,
-                &routing_service.earning_wallet,
-            );
-        })
+
+        let new_postings = NewPostingsDebugContainer::new(&self.logger)
+            .add(msg.exit.earning_wallet.address(), exit_sum);
+
+        let new_postings = self.handle_routing_services_consumed(msg, msg_id, new_postings);
+
+        self.logging_utils.accounting_msgs_stats.manage_debug_log(
```

Comment:
I'd prefer if you directly log it here... that'll be clearer. From the outside, it only appears it's just a log, but after a deep dive, I see use of `&mut`, and I don't think it's ideal to do it this way, we should do mutations and logging separately, using different functions step by step. Give it a thought. I didn't give an absolute example, because there is nesting in the code and you would need to refactor on multiple levels to fix that. I'd appreciate if you do.